src/main/java/ai/chalk/internal/Utils.java

```diff
@@ -1,16 +1,31 @@
 package ai.chalk.internal;
 
+import ai.chalk.features.Name;
+
 import java.lang.reflect.Array;
 import java.lang.reflect.Field;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
 import java.time.Duration;
-import java.time.temporal.ChronoUnit;
 import java.util.List;
+import java.time.temporal.ChronoUnit;
 
 public class Utils {
-    public static String toSnakeCase(String s) {
-        return s.replaceAll("([a-z])([A-Z])", "$1_$2").toLowerCase();
+    public static String getResolvedName(Field field) {
+        // If has the Name annotation, use that as the name
+        // Otherwise, use the field name snake cased
+        if (field.isAnnotationPresent(Name.class)) {
+            return field.getAnnotation(Name.class).value();
+        }
+        return chalkpySnakeCase(field.getName());
+
+    }
+    public static String chalkpySnakeCase(String s) {
+        // Aims to be in parity with chalkpy's impl
+        s = s.replaceAll("(.)([A-Z][a-z]+)", "$1_$2");
```

Comment:
I'm sure you've thought deeply about the order of the replaceAlls but just in case you didn't: I think this is mostly correct (i'm not sure what our intended behavior for features that, for example, start with `_`, which to my understanding is already not good practice for Java)

Reply:
Yeah I think here we want absolute parity with chalkpy, so I think we need to keep this even though it doesn't handle getting rid of leading underscores (if that's what you were thinking)